src/main/java/org/moditect/jfrunit/JfrEvents.java

```diff
@@ -305,4 +326,19 @@ private String getDumpFileName() {
     private String getDefaultDumpFileName() {
         return testMethod.getDeclaringClass().getName() + "-" + testMethod.getName() + ".jfr";
     }
+
+    public JfrAnalysisResults automaticAnalysis() {
+        if (analysis == null) {
+            try {
+                Path recordingPath = getRecordingFilePath();
+                dumpRecording(recordingPath);
+
+                analysis = new JfrAnalysisResults(JfrAnalysis.analysisRecording(recordingPath.toAbsolutePath().toString(), Severity.INFO));
+            }
+            catch (IOException | URISyntaxException e) {
+                LOGGER.log(Level.WARNING, "Unable to analyse jfr recording: " + e.getLocalizedMessage());
```

Comment:
Perhaps this should fail immediately with runtime exception, rather than users having assert on the non null result (not a logic in test) or NPEs later. I assume this would be rare, wouldn't it?

Reply:
Yes, this should be rare, I will change to a RuntimeException

Reply:
I have changed this to throw a RuntimeException